packages/expo-asset/src/AssetSourceResolver.web.ts

```diff
@@ -52,54 +31,47 @@ export default class AssetSourceResolver {
     jsbundleUrl: string | undefined | null,
     asset: PackagerAsset
   ) {
+    if (!serverUrl) {
+      throw new Error('Web assets require a server URL');
+    }
+
     this.serverUrl = serverUrl;
-    this.jsbundleUrl = jsbundleUrl;
+    this.jsbundleUrl = null;
     this.asset = asset;
   }
+
+  // Always true for web runtimes
   isLoadedFromServer(): boolean {
-    return !!this.serverUrl;
+    return true;
   }
+
+  // Always false for web runtimes
   isLoadedFromFileSystem(): boolean {
-    return !!(this.jsbundleUrl && this.jsbundleUrl.startsWith('file://'));
+    return false;
   }
-  defaultAsset(): ResolvedAssetSource {
-    if (this.isLoadedFromServer()) {
-      return this.assetServerURL();
-    }
 
-    return this.scaledAssetURLNearBundle();
+  defaultAsset(): ResolvedAssetSource {
+    return this.assetServerURL();
   }
+
+  /**
+   * @returns absolute remote URL for the hosted asset.
+   */
   assetServerURL(): ResolvedAssetSource {
-    invariant(!!this.serverUrl, 'need server to load from');
-    return this.fromSource(
-      this.serverUrl +
-        getScaledAssetPath(this.asset) +
-        '?platform=' +
-        Platform.OS +
-        '&hash=' +
-        this.asset.hash
-    );
-  }
-  scaledAssetPath(): ResolvedAssetSource {
-    return this.fromSource(getScaledAssetPath(this.asset));
-  }
-  scaledAssetURLNearBundle(): ResolvedAssetSource {
-    const path = this.jsbundleUrl || '';
-    return this.fromSource(path + getScaledAssetPath(this.asset));
-  }
-  resourceIdentifierWithoutScale(): ResolvedAssetSource {
-    throw new UnavailabilityError('react-native', 'resourceIdentifierWithoutScale()');
-  }
-  drawableFolderInBundle(): ResolvedAssetSource {
-    throw new UnavailabilityError('react-native', 'drawableFolderInBundle()');
+    const fromUrl = new URL(this.serverUrl);
+    fromUrl.pathname = getScaledAssetPath(this.asset);
```

Comment:
Should this preserve the pathname in the serverUrl? For instance, if serverUrl is "https://example.com/hello", then should an asset at "icon.png" or "./icon.png" resolve to "https://example.com/hello/icon.png" or "https://example.com/icon.png"?

```suggestion
    // conceptually speaking (path.join() is a Node function)
    fromUrl.pathname = path.join(fromUrl.pathname, getScaledAssetPath(this.asset));
```

The main use case I can think of for this is using a cloud storage bucket, which often has a base URL like "region.cloudproviderstorage.com/bucket".

Or do we expect paths in the base URL to be handled by `asset.httpServerLocation`?